br/pkg/streamhelper/advancer_test.go

```diff
@@ -604,18 +686,74 @@ func TestUnregisterAfterPause(t *testing.T) {
 		c.CheckPointLagLimit = 1 * time.Minute
 	})
 	adv.StartTaskListener(ctx)
+
+	// No matter how many times the task is paused, after put a new one the task should run normally
+	// First sequence: pause -> unregister -> put
```

Comment:
The key function of the test is to check that even after the `isPaused` is set, when unregister it will be erased.

But seems in this new version, the `isPaused` is never set?